packages/workspace/src/workspace/reference_indexes.ts

```diff
@@ -89,17 +101,15 @@ const updateUniqueIndex = async (
 const getReferenceTargetIndexUpdates = (
   change: Change<Element>,
   changeToReferences: Record<string, ChangeReferences>,
-): RemoteMapEntry<ElemID[]>[] => {
-  const indexUpdates: RemoteMapEntry<ElemID[]>[] = []
-
-  const references = changeToReferences[getChangeData(change).elemID.getFullName()]
-    .currentAndNew
-  const baseIdToReferences = _(references)
-    .groupBy(reference => reference.referenceSource.createBaseID().parent.getFullName())
-    .mapValues(referencesGroup => referencesGroup.map(ref => ref.referenceTarget))
-    .value()
+): RemoteMapEntry<collections.treeMap.TreeMap<ElemID>>[] => {
+  const indexUpdates: RemoteMapEntry<collections.treeMap.TreeMap<ElemID>>[] = []
 
   if (isObjectTypeChange(change)) {
+    const references = changeToReferences[getChangeData(change).elemID.getFullName()]
+      .currentAndNew
+    const baseIdToReferences = _(references)
+      .groupBy(reference => reference.referenceSource.createBaseID().parent.getFullName())
+      .value()
```

Comment:
```suggestion
    const baseIdToReferences = _.groupBy(
      references,
      reference => reference.referenceSource.createBaseID().parent.getFullName()
    )
```

or even
```suggestion
    const baseIdToReferences = _.groupBy(
      changeToReferences[getChangeData(change).elemID.getFullName()].currentAndNew,
      reference => reference.referenceSource.createBaseID().parent.getFullName()
    )
```

or better yet, move the `const elemId = getChangeData(change).elemID.getFullName()` line to the top and then

```suggestion
    const baseIdToReferences = _.groupBy(
      changeToReferences[elemId].currentAndNew,
      reference => reference.referenceSource.createBaseID().parent.getFullName()
    )
```